### PR TITLE
[3.x] Add OS.get_external_data_dir() to get Android external directory

### PR DESCRIPTION
**Pull request by madmiraal**
3.x version of #49435.

### CODE REVIEW

File: platform/android/os_android.cpp [hunk 1]
```diff
@@ -781,6 +781,27 @@ String OS_Android::get_user_data_dir() const {
...
+		chdir(data_dir.utf8().get_data());
+		// Get the actual directory without the potential symlink.
+		char data_current_dir_name[2048];
+		getcwd(data_current_dir_name, 2048);
+		// Convert back to a String.
+		data_dir.parse_utf8(data_current_dir_name);
+		// Restore original current directory.
+		chdir(current_dir_name);
+		return data_dir;
+	}
```
Comment (m4gr3d) [inline platform/android/os_android.cpp:801]: This is the same logic as in `get_user_data_dir()`. Can you wrap the logic into a util function and have both functions call it instead.

**Review by m4gr3d — CHANGES_REQUESTED** [review]
Looks great! Same feedback as for #49435

File: platform/android/os_android.cpp (hunk 1 above)
Comment (madmiraal) [reply platform/android/os_android.cpp]: Done.

**Review by m4gr3d — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**